### PR TITLE
Adds support for 8 stringr functions

### PR DESCRIPTION
**Pull request by edgararuiz**
Adding support for stringr functions: `str_length()`, `str_to_upper()`, `str_to_lower()`,
  `str_locate()`, `str_replace_all()`, `str_detect()`, `str_trim()`, `str_replace_na()`

Live database test results are below.  

- `str_locate()` fails because it return a single value of the beginning location of the searched string instead of a begin-end vector
- `str_detect()` will only work with `filter()` for now.  I think we can implement a different solution once the MSSQL is.na PR is merged

mutate

Filter results below, they are all red, but notice that the result for all of them is **Passed**

filter

cc @javierluraschi

### CODE REVIEW

**Comment by hadley** [comment]
Next time,  add `scale_fill_colour(limits = c("Failed", "Passed"))` to avoid the problem in the second plot.

**Comment by edgararuiz** [comment]
Ah! Thanks, I will

File: R/db-odbc-mssql.R [hunk 1]
```diff
@@ -70,7 +70,19 @@
                             "LTRIM(RTRIM(", x ,"))"
                           )},
                       # MSSQL supports CONCAT_WS in the CTP version of 2016
-      paste         = sql_not_supported("paste()")
+      paste         = sql_not_supported("paste()"),
```
[the remark this answers is not in the document]
Comment (edgararuiz) [reply R/db-odbc-mssql.R]: `CONCAT_WS`  is supported by starting with version 14 (MS SQL 2017) (https://docs.microsoft.com/en-us/sql/t-sql/functions/concat-ws-transact-sql).   I figured most companies would not have this version yet, so I made the call to mark it at unsupported to reduce questions and frustrations.  

An idea would be to use the MS SQL version usually passed in the connection info to determine if to support the function or pass the error message.

[the remark this answers is not in the document]
Comment (hadley) [reply R/db-odbc-mssql.R]: Ok, let's try doing that but in a separate PR.

[the remark this answers is not in the document]
Comment (edgararuiz) [reply R/db-odbc-mssql.R]: Sounds good

**Comment by hadley** [comment]
Thanks!

**Pull request merged**